### PR TITLE
Configure: Fix swapped responses for SETLOCALE_ACCEPTS_ANY_LOCALE_NAME

### PR DESCRIPTION
**Pull request by khwilliamson**
Answering y gave the opposite of what was intended; and vice-versa

### CODE REVIEW

**Comment by mauke** [comment]
I suggest rewording the question, too. From
```sh
"When you set your locale to something besides C$c_utf8_locale, does it do so, or just pretend to?"
```
to something like
```sh
"When you set your locale to something besides C$c_utf8_locale, does it actually do so (and not just pretend to)?"
```
in order to make it clear what a "yes" answer represents.

**Review by tonycoz — APPROVED** [review]

**Comment by khwilliamson** [comment]
I haven't pushed this because mauke found some other issues.

**Comment by Tux** [comment]
What is the state now?

**Comment by jkeenan** [comment]
> I haven't pushed this because mauke found some other issues.

@khwilliamson , @mauke, can we get an update on the state of this merge request?  Thanks.

**Comment by mauke** [comment]
I think the other issues is #21406.

**Comment by khwilliamson** [comment]
This fixes #21406.  I found the other issues through #irc scrollback, and wrote https://github.com/Perl/perl5/issues/21715
